components/script/dom/serviceworkercontainer.rs

```diff
@@ -56,7 +65,27 @@ impl ServiceWorkerContainerMethods for ServiceWorkerContainer {
     // https://slightlyoff.github.io/ServiceWorker/spec/service_worker/#service-worker-container-register-method
     fn Register(&self,
                 script_url: USVString,
-                options: &RegistrationOptions) -> Fallible<Root<ServiceWorkerRegistration>> {
+                options: &RegistrationOptions) -> Promise<ServiceWorker> {
+        
+        
+        /*Let p be a promise.
+        Let client be the context object’s service worker client.
+        Let scriptURL be the result of parsing scriptURL with the context object’s relevant settings object’s API base URL.
+        Let scopeURL be null.
+        If options.scope is present, set scopeURL to the result of parsing options.scope with the context object’s relevant settings object’s API base URL.
+        Invoke Start Register with scopeURL, scriptURL, p, client, client’s creation URL and options.type.
+        Return p.    */ 
```

Comment:
As mentioned in #13409, I don't think there's any need to focus on the specification. It's enough to know that it says to create a promise and return it, and either resolve it with a successful service worker or else reject it with an error (instead of returning error values directly).